from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_HowToCreateAFile(object):
    def setupUi(self, HowToCreateAFile):
        HowToCreateAFile.setObjectName("HowToCreateAFile")
        HowToCreateAFile.resize(1100, 600)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("../Facultate/Licenta/dot-crossed_icon-icons.com_68558.ico"), QtGui.QIcon.Normal,
                       QtGui.QIcon.Off)
        HowToCreateAFile.setWindowIcon(icon)

        with open('HowToCreateAFile.qss', 'r') as f:
            style = f.read()
            HowToCreateAFile.setStyleSheet(style)

        self.centralwidget = QtWidgets.QWidget(HowToCreateAFile)
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.centralwidget)
        self.verticalLayout.setObjectName("verticalLayout")
        self.mainFrame = QtWidgets.QFrame(self.centralwidget)
        self.mainFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.mainFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.mainFrame.setObjectName("mainFrame")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.mainFrame)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.topFrame = QtWidgets.QFrame(self.mainFrame)
        self.topFrame.setMaximumSize(QtCore.QSize(16777215, 100))
        self.topFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.topFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.topFrame.setObjectName("topFrame")
        self.verticalLayout_3 = QtWidgets.QVBoxLayout(self.topFrame)
        self.verticalLayout_3.setObjectName("verticalLayout_3")
        self.titleLabel = QtWidgets.QLabel(self.topFrame)
        self.titleLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.titleLabel.setObjectName("titleLabel")
        self.verticalLayout_3.addWidget(self.titleLabel)
        self.verticalLayout_2.addWidget(self.topFrame)
        self.headFrame = QtWidgets.QFrame(self.mainFrame)
        self.headFrame.setMaximumSize(QtCore.QSize(16777215, 100))
        self.headFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.headFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.headFrame.setObjectName("headFrame")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(self.headFrame)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.targetNameLabel = QtWidgets.QLabel(self.headFrame)
        self.targetNameLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.targetNameLabel.setObjectName("targetNameLabel")
        self.horizontalLayout_2.addWidget(self.targetNameLabel)
        self.targetLabel = QtWidgets.QLabel(self.headFrame)
        self.targetLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.targetLabel.setObjectName("targetLabel")
        self.horizontalLayout_2.addWidget(self.targetLabel)
        self.firstFeatureLabel = QtWidgets.QLabel(self.headFrame)
        self.firstFeatureLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.firstFeatureLabel.setObjectName("firstFeatureLabel")
        self.horizontalLayout_2.addWidget(self.firstFeatureLabel)
        self.secondFeatureLabel = QtWidgets.QLabel(self.headFrame)
        self.secondFeatureLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.secondFeatureLabel.setObjectName("secondFeatureLabel")
        self.horizontalLayout_2.addWidget(self.secondFeatureLabel)
        self.dotsLabel = QtWidgets.QLabel(self.headFrame)
        self.dotsLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.dotsLabel.setObjectName("dotsLabel")
        self.horizontalLayout_2.addWidget(self.dotsLabel)
        self.lastFeatureLabel = QtWidgets.QLabel(self.headFrame)
        self.lastFeatureLabel.setAlignment(QtCore.Qt.AlignCenter)
        self.lastFeatureLabel.setObjectName("lastFeatureLabel")
        self.horizontalLayout_2.addWidget(self.lastFeatureLabel)
        self.verticalLayout_2.addWidget(self.headFrame)
        self.cellsExplainedFrame = QtWidgets.QFrame(self.mainFrame)
        self.cellsExplainedFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.cellsExplainedFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.cellsExplainedFrame.setObjectName("cellsExplainedFrame")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout(self.cellsExplainedFrame)
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        self.targetNoteLabel = QtWidgets.QLabel(self.cellsExplainedFrame)
        self.targetNoteLabel.setObjectName("targetNoteLabel")
        self.verticalLayout_4.addWidget(self.targetNoteLabel)
        self.targetNameNoteLabel = QtWidgets.QLabel(self.cellsExplainedFrame)
        self.targetNameNoteLabel.setObjectName("targetNameNoteLabel")
        self.verticalLayout_4.addWidget(self.targetNameNoteLabel)
        self.featuresNoteLabel = QtWidgets.QLabel(self.cellsExplainedFrame)
        self.featuresNoteLabel.setObjectName("featuresNoteLabel")
        self.verticalLayout_4.addWidget(self.featuresNoteLabel)
        self.verticalLayout_2.addWidget(self.cellsExplainedFrame)
        self.noticeLabel = QtWidgets.QLabel(self.mainFrame)
        self.noticeLabel.setMaximumSize(QtCore.QSize(16777215, 50))
        self.noticeLabel.setObjectName("noticeLabel")
        self.verticalLayout_2.addWidget(self.noticeLabel)
        self.notesFrame = QtWidgets.QFrame(self.mainFrame)
        self.notesFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.notesFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.notesFrame.setObjectName("notesFrame")
        self.verticalLayout_6 = QtWidgets.QVBoxLayout(self.notesFrame)
        self.verticalLayout_6.setObjectName("verticalLayout_6")
        self.noteLabel_4 = QtWidgets.QLabel(self.notesFrame)
        self.noteLabel_4.setObjectName("noteLabel_4")
        self.verticalLayout_6.addWidget(self.noteLabel_4)
        self.noteLabel_5 = QtWidgets.QLabel(self.notesFrame)
        self.noteLabel_5.setObjectName("noteLabel_5")
        self.verticalLayout_6.addWidget(self.noteLabel_5)
        self.noteLabel_6 = QtWidgets.QLabel(self.notesFrame)
        self.noteLabel_6.setObjectName("noteLabel_6")
        self.verticalLayout_6.addWidget(self.noteLabel_6)
        self.verticalLayout_2.addWidget(self.notesFrame)
        self.verticalLayout.addWidget(self.mainFrame)
        HowToCreateAFile.setCentralWidget(self.centralwidget)

        self.retranslateUi(HowToCreateAFile)
        QtCore.QMetaObject.connectSlotsByName(HowToCreateAFile)

    def retranslateUi(self, HowToCreateAFile):
        _translate = QtCore.QCoreApplication.translate
        HowToCreateAFile.setWindowTitle(_translate("HowToCreateAFile", "SVM Visualizer"))
        self.mainFrame.setAccessibleName(_translate("HowToCreateAFile", "mainFrame"))
        self.topFrame.setAccessibleName(_translate("HowToCreateAFile", "topFrame"))
        self.titleLabel.setAccessibleName(_translate("HowToCreateAFile", "titleLabel"))
        self.titleLabel.setText(_translate("HowToCreateAFile", "SVM Visualizer"))
        self.headFrame.setAccessibleName(_translate("HowToCreateAFile", "headFrame"))
        self.targetNameLabel.setAccessibleName(_translate("HowToCreateAFile", "targetNameLabel"))
        self.targetNameLabel.setText(_translate("HowToCreateAFile", "TargetName"))
        self.targetLabel.setAccessibleName(_translate("HowToCreateAFile", "targetLabel"))
        self.targetLabel.setText(_translate("HowToCreateAFile", "Target"))
        self.firstFeatureLabel.setAccessibleName(_translate("HowToCreateAFile", "firstFeatureLabel"))
        self.firstFeatureLabel.setText(_translate("HowToCreateAFile", "FirstFeature"))
        self.secondFeatureLabel.setAccessibleName(_translate("HowToCreateAFile", "secondFeatureLabel"))
        self.secondFeatureLabel.setText(_translate("HowToCreateAFile", "SecondFeature"))
        self.dotsLabel.setAccessibleName(_translate("HowToCreateAFile", "dotsLabel"))
        self.dotsLabel.setText(_translate("HowToCreateAFile", "..."))
        self.lastFeatureLabel.setAccessibleName(_translate("HowToCreateAFile", "lastFeatureLabel"))
        self.lastFeatureLabel.setText(_translate("HowToCreateAFile", "LastFeature"))
        self.cellsExplainedFrame.setAccessibleName(_translate("HowToCreateAFile", "cellsExplainedFrame"))
        self.targetNoteLabel.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.targetNoteLabel.setText(_translate("HowToCreateAFile", "● Target - the value of the classes (0 and 1)"))
        self.targetNameNoteLabel.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.targetNameNoteLabel.setText(_translate("HowToCreateAFile", "● TargetName - column that contains the names of the classes that each value belongs to"))
        self.featuresNoteLabel.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.featuresNoteLabel.setText(_translate("HowToCreateAFile", "● FirstFeature, SecondFeature, ... - columns with the values of each features"))
        self.noticeLabel.setAccessibleName(_translate("HowToCreateAFile", "noticeLabel"))
        self.noticeLabel.setText(_translate("HowToCreateAFile", "Things to keep in mind when creating a file for SVM Visualizer:"))
        self.notesFrame.setAccessibleName(_translate("HowToCreateAFile", "notesFrame"))
        self.noteLabel_4.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.noteLabel_4.setText(_translate("HowToCreateAFile", "● The name of each of the header cells must not contain spaces "))
        self.noteLabel_5.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.noteLabel_5.setText(_translate("HowToCreateAFile", "● \"TargetName\" and \"Target\" must be present in order for the program to work properly"))
        self.noteLabel_6.setAccessibleName(_translate("HowToCreateAFile", "noteLabel"))
        self.noteLabel_6.setText(_translate("HowToCreateAFile", "● The order of the columns must be respected in order for the program to work properly"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    HowToCreateAFile = QtWidgets.QMainWindow()
    ui = Ui_HowToCreateAFile()
    ui.setupUi(HowToCreateAFile)
    HowToCreateAFile.show()
    sys.exit(app.exec_())
